app/src/main/java/com/programmers/library/Mode.java

```diff
@@ -0,0 +1,41 @@
+package com.programmers.library;
+
+import com.programmers.library.controller.Controller;
+import com.programmers.library.repository.FileRepository;
+import com.programmers.library.repository.TestRepository;
+import com.programmers.library.service.Service;
+
+public enum Mode {
+    NORMAL(1, "일반"),
+    TEST(2, "테스트");
+
+    private final int id;
+    private final String name;
+    private final Controller controller;
+
+    Mode(int id, String name) {
+        this.id = id;
+        this.name = name;
+        this.controller = id == 2 ? new Controller(new Service(new TestRepository())) : new Controller(new Service(new FileRepository()));
+    }
+
+    public void run() {
+        System.out.println("[System] " + name + " 모드로 애플리케이션을 실행합니다.");
+        controller.run();
+    }
+
+    public static void showModes() {
+        System.out.println("Q. 모드를 선택해주세요.");
+        for (Mode mode : Mode.values()) {
+            System.out.println((mode.id + ". " + mode.name));
+        }
+        System.out.print("> ");
+    }
+
+    public static Mode selectMode(int id) {
+        if (id == 2) {
+            return Mode.TEST;
+        }
+        return Mode.NORMAL;
+    }
```

Comment:
```suggestion
    public static Mode selectMode(int id) {
        return Arrays.stream(values())
                .filter(mode -> mode.id == id)
                .findFirst()
                .orElse(NORMAL);
    }
```

현재는 두 개의 타입이 정의되어 있기 때문에 걱정 없지만 타입이 더 늘어나는 상황이라면 조건문을 추가하지 않고 선택하는 방법으로 구현하는 것을 고민해보는 것도 좋을 것 같습니다.